controllers/topology/reconcile_state_test.go

```diff
@@ -203,6 +203,259 @@ func TestReconcileInfrastructureCluster(t *testing.T) {
+	bootstrapTemplate4WithChanges.SetLabels(map[string]string{"foo": "bar"})
+	md4WithRotatedTemplates := newFakeMachineDeploymentTopologyState("md-4", infrastructureMachineTemplate4WithChanges, bootstrapTemplate4WithChanges)
+
+	infrastructureMachineTemplate5 := newFakeInfrastructureMachineTemplate(metav1.NamespaceDefault, "infrastructure-machine-5").Obj()
+	bootstrapTemplate5 := newFakeBootstrapTemplate(metav1.NamespaceDefault, "bootstrap-config-5").Obj()
+	md5 := newFakeMachineDeploymentTopologyState("md-5", infrastructureMachineTemplate5, bootstrapTemplate5)
+	infrastructureMachineTemplate5WithChangedKind := infrastructureMachineTemplate5.DeepCopy()
+	infrastructureMachineTemplate5WithChangedKind.SetKind("ChangedKind")
+	md5WithChangedInfrastructureMachineTemplateKind := newFakeMachineDeploymentTopologyState("md-4", infrastructureMachineTemplate5WithChangedKind, bootstrapTemplate5)
+
+	infrastructureMachineTemplate6 := newFakeInfrastructureMachineTemplate(metav1.NamespaceDefault, "infrastructure-machine-6").Obj()
+	bootstrapTemplate6 := newFakeBootstrapTemplate(metav1.NamespaceDefault, "bootstrap-config-6").Obj()
+	md6 := newFakeMachineDeploymentTopologyState("md-6", infrastructureMachineTemplate6, bootstrapTemplate6)
+	bootstrapTemplate6WithChangedNamespace := bootstrapTemplate6.DeepCopy()
+	bootstrapTemplate6WithChangedNamespace.SetNamespace("ChangedNamespace")
+	md6WithChangedBootstrapTemplateNamespace := newFakeMachineDeploymentTopologyState("md-6", infrastructureMachineTemplate6, bootstrapTemplate6WithChangedNamespace)
+
+	infrastructureMachineTemplate7 := newFakeInfrastructureMachineTemplate(metav1.NamespaceDefault, "infrastructure-machine-7").Obj()
+	bootstrapTemplate7 := newFakeBootstrapTemplate(metav1.NamespaceDefault, "bootstrap-config-7").Obj()
+	md7 := newFakeMachineDeploymentTopologyState("md-7", infrastructureMachineTemplate7, bootstrapTemplate7)
+
+	tests := []struct {
+		name                                      string
+		current                                   []machineDeploymentTopologyState
+		desired                                   []machineDeploymentTopologyState
+		want                                      []machineDeploymentTopologyState
+		wantInfrastructureMachineTemplateRotation map[string]bool
+		wantBootstrapTemplateRotation             map[string]bool
+		wantErr                                   bool
+	}{
+		{
+			name:    "Should create desired MachineDeployment if the current does not exists yet",
+			current: nil,
+			desired: []machineDeploymentTopologyState{md1},
+			want:    []machineDeploymentTopologyState{md1},
+			wantErr: false,
+		},
+		{
+			name:    "No-op if current MachineDeployment is equal to desired",
+			current: []machineDeploymentTopologyState{md1},
+			desired: []machineDeploymentTopologyState{md1},
+			want:    []machineDeploymentTopologyState{md1},
+			wantErr: false,
+		},
+		{
+			name:    "Should update MachineDeployment with InfrastructureMachineTemplate rotation",
```

Comment:
Let's have a similar test for bootstrap template

Reply:
Imho I already have one directly below (`"Should update MachineDeployment with BootstrapTemplate rotation"`)

(don't ask my why gofmt formats the next test case that so differently :))